File: Add Code Here/C/CountSort.c
#include<stdlib.h>
#include<stdio.h>


void display_array(int size, int a[]){
    for (int i = 0; i < size; i++)
    {
        printf("%d   ", a[i]);
    }
    printf("\n");
}

int max_element(int array[], int count){
    int max=0;
    for (int i = 0; i < count; i++)
    {
        
        if(array[i]>max)
            max=array[i];
    }
    return max;
}

int * countsort(int A[], int B[], int size){
    int max=max_element(A, size);
    int C[max+1];
    for(int i=0; i<=max; i++){
        C[i]=0;
    }
    display_array(max+1, C);
    for (int i = 0; i <size; i++)
    {
        C[A[i]]=C[A[i]]+1;
    }
    display_array(max+1, C);
    for(int i=1; i<=max; i++){
        C[i]=C[i]+C[i-1];
    }
    display_array(max+1, C);
    for (int i = size-1; i>=0; i--)
    {
        B[C[A[i]]-1]=A[i];
        C[A[i]]-=1;
    }
    return B;
}
int main(){
    printf("Enter array size    :\n");
    int size;
    scanf("%d", &size);
    int array[size];
    printf("Enter elements of array  :\n");
    for (int i = 0; i < size; i++)
    {
        scanf("%d", &array[i]);
    }
    display_array(size, array);
    int B[size];
    int *c=countsort(array, B, size);
    display_array(size, c);

}

File: Add Code Here/Java/count_sort.java
package Sorting;

import java.util.Scanner;

public class count_sort {
    static void countsort(int arr[], int n, int max)
    {
        int i;
        int []sort = new int[10];
        int []count = new int[10];

        for(i=0;i<=max;i++){
            count[i]=0;
        }
        for(i=0;i<n;i++){
            count[arr[i]]++;
        }
        for(i=1;i<=max;i++){
            count[i]=count[i]+count[i-1];
        }
        for(i=n-1;i>=0;i--){
            sort[count[arr[i]]-1]=arr[i];
            count[arr[i]]--;
        }

        for(i=0;i<n;i++)
            arr[i]=sort[i];

    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the size of array: ");
        int n = input.nextInt();
        int [] arr = new int[n];
        int i, max=0;
        System.out.println("Enter "+ n + " element in array: ");
        for(i=0;i<n;i++) {
            arr[i] = input.nextInt();
            if (arr[i] > max) {
                max = arr[i];
            }
        }
        countsort(arr, n, max);
        System.out.print("Sorted array: ");
        for(i=0;i<n;i++)
            System.out.print(arr[i]+ " ");
    }
}
